Tests/Tests/IncrementalStore.m

```diff
@@ -731,6 +734,144 @@ - (void)testFetchConstraints
     }
 }
 
+- (void)testBatchUpdates
+{
+    int num_entries = 100;
+
+    XCTAssertTrue((num_entries % 4) == 0, @"Test assumes num_entries is mod 4: %d", num_entries);
+
+    NSError *err = nil;
+    // This will create the database
+    NSManagedObjectContext *moc = self.managedObjectContext;
+    XCTAssertNotNil(moc, @"could not create Context");
+
+    for (int i = 0; i < num_entries; i++) {
+        Entry *e = MakeEntry(moc);
+        e.text = NSStringFromSelector(_cmd);
+        e.check = (i % 2) ? @NO : @YES;
+        e.i64 = @(i);
+    }
+
+    // push it out
+    XCTAssertTrue([moc save:&err], @"Save Failed: %@", err);
+
```

Comment:
What I'd like to do is defer this until you merge the query pull request, because this has the CoreData path in the tests that I can use to verify our behavior against that.